### ISSUE TITLE
Improve error message when assigning wrong value (assigning empty array for Observable typed var)

### ISSUE THREAD

**Issue by boazrymland** [microsoft/TypeScript#15765]
**TypeScript Version:**  2.2.2

**Code**

```ts
@Injectable()
export class ConfigurationManagerService {
  sutFilesObservable: Observable<{}[]> = [];

  constructor() {
  }
}

```

**Actual behavior:**
The code above is buggy since it initializes the sutFilesObservable property with incorrectly typed value (array, while this property is of type Observable). 
Yet, the error message is not clear: "Type 'undefined[]' is not assignable to type 'Observable<{}[]>'. Property '_isScalar' is missing in type 'undefined[]'".

**Expected behavior:**
An error should be flagged but it should be more clear, something along: "Type [] is not assignable to type 'Observable<{}[]>'". If the 'undefined' is needed I think its ok, but the second sentence ("Property 'isScalar'...") is only distracting and confusing and should preferably be removed.

**Comment by mhegazy** [comment]
The TS type system is structural, so comparing two types involve walking their properties and comparing them recursively. these error messages show you the path the compiler took to reach the conclusion that these two types are not structurally compatible. 
In this case it might have been less noisy, but it is hard to see how this can be optimized for the general case.
That said, we are definitely open to suggestions on how to make this better.

**Comment by boazrymland** [comment]
I understand that you want to keep the technical details in the message. This do come with a cost though, cost of obscure message that is unclear to less then _expert_ typescript developer. This is a challenge.

I can think of two possible solutions for this, to enjoy both worlds, but to put some order in the message that will help distinguish between the uber-technical and the to-the-masses message:
1. Have some flag for tsc, which will serve as kind of 'message level'. By default, it should be off or otherwise indicating low verbosity is requested (since most developers are _not_ typescript experts). In that case, the error that should have been emitted was the first sentence only (without the "property _isScalar...").When on or otherwise indicating high verbosity is requested, it would be as above.

2. This could augment the first suggestion: In the desired error message in tsc engine, distinguish between the 'regular' and 'detailed' parts. Then, put the 'detailed' part in parenthesis, prefixed by some clear text explaining that the text ahead might be an overkill to most readers, for example: "(technical info: <here comes the technical detail>)". 
This can be used in conjunction with the first suggestion above so if the extra verbosity is required, it is provided as suggested in this option.

Thanks :)
